### PR TITLE
Use a filtered informer factory, refactor NewController

### PR DESCRIPTION
**Pull request by copejon**
Switch to filtered informers, move controller responsibilities into NewController, add selector labels

**This breaks unit tests on the controller.**

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -23,27 +25,49 @@ const (
...
-		pvcInformer:    pvcInformer,
-		pvcListWatcher: pvcListWatcher,
-		importerImage:  importerImage,
+func NewController(client kubernetes.Interface, importerImage string) *Controller {
+
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	// Setup all informers to only watch objects with the LabelSelector `app=containerized-data-importer`
+	informerFactory := informers.NewFilteredSharedInformerFactory(client, 10*time.Minute, "", func(options *metav1.ListOptions) {
+		options.LabelSelector = "app=containerized-data-importer"
```
Comment (jeffvance) [inline pkg/controller/controller.go]: do we want to make this key and value a constant?

File: pkg/controller/controller_test.go [hunk 2]
```diff
@@ -56,7 +57,9 @@ var _ = Describe("Controller", func() {
 		pvcInformer := cache.NewSharedIndexInformer(objSource, pvc, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
 		queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
 		pvcListWatcher := k8stesting.NewFakeControllerSource()
-		controller = NewController(fakeClient, queue, pvcInformer, pvcListWatcher, importerTag)
+		var err error
+		controller, err = NewController(fakeClient, common.CDI_SELECTOR_LABEL,importerTag)
+		if error
```
Comment (jeffvance) [inline pkg/controller/controller_test.go]: Replace this with `Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("setUpInformer: NewController error: %v", err))`

File: pkg/controller/controller_test.go [hunk 3]
```diff
@@ -56,7 +57,9 @@ var _ = Describe("Controller", func() {
 		pvcInformer := cache.NewSharedIndexInformer(objSource, pvc, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
 		queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
 		pvcListWatcher := k8stesting.NewFakeControllerSource()
-		controller = NewController(fakeClient, queue, pvcInformer, pvcListWatcher, importerTag)
+		var err error
+		controller, err = NewController(fakeClient, common.CDI_SELECTOR_LABEL,importerTag)
```
Comment (jeffvance) [inline pkg/controller/controller_test.go]: replace `importerTag` with `common.IMPORTER_DEFAULT_IMAGE`

**Comment by copejon** [comment]
Moved branch into UPSTREAM.

**Pull request closed without merging**